File: fe/src/types/UserToProjectSchema.ts

import { UserDetail } from "./UserProperties";

export interface UserToProjectFull {
    userId?: string;
    projectId?: string;
    role?: string;
    active?: boolean;
    datCreation?: string;
    users?: Array<UserDetail>;
}

export interface ProjectFull {
    projectName?: string;
    descrizioneProgetto?: string;
    datInizio?: string;
    datFine?: string;
    stato?: string;
    avanzamento?: number;
    costo?: number;
    userToProjects?: Array<UserToProjectFull>;
    id: string;
}

File: be/persistence/schemas/UserToProjectSchema.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from .UserSchema import UserBase

class UserToProjectBase(BaseModel):
    userId: Optional[str] = None
    projectId: Optional[str] = None
    role: Optional[str] = None
    active: Optional[bool] = None
    datCreation: Optional[datetime] = None
    avanzamento: Optional[int] = None

class UserToProjectCreate(UserToProjectBase):
    pass

class UserToProjectResponse(UserToProjectBase):
    id: str

    class Config:
        from_attributes = True

class UserToProjectFull(UserToProjectBase):
    users: Optional[List[UserBase]] = None